driver-core/src/test/java/com/datastax/driver/core/AsyncQueryTest.java

```diff
@@ -76,19 +90,20 @@ public void should_init_cluster_and_session_if_needed() throws Exception {
             // Neither cluster2 nor session2 are initialized at this point
             assertThat(cluster2.manager.metadata).isNull();
 
-            ResultSetFuture future = session2.executeAsync(String.format("select v from %s.foo where k = 1", keyspace));
+            ResultSetFuture future = session2.executeAsync("select release_version from system.local");
             Row row = Uninterruptibles.getUninterruptibly(future).one();
 
-            assertThat(row.getInt(0)).isEqualTo(1);
+            assertThat(row.getString(0)).isNotEmpty();
         } finally {
             if (cluster2 != null)
                 cluster2.close();
         }
     }
 
-    @Test(groups = "short")
-    public void should_chain_query_on_async_session_init() throws Exception {
-        ListenableFuture<Integer> resultFuture = connectAndQuery(this.keyspace);
+    @Test(groups = "short", dataProvider = "keyspace", enabled = false,
+        description = "disabled because the blocking USE call in the current pool implementation makes it deadlock")
+    public void should_chain_query_on_async_session_init(String keyspace) throws Exception {
```

Comment:
Yes it is. There's still value in testing with the same executor, but we can do both.